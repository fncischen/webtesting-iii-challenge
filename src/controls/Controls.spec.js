// Test away!

import React from 'react';
import renderer from 'react-test-renderer';
import Controls from "./Controls";


describe("Controls", () => { 

it('Control button is toggled for locked and closed', () => {
  const tree = renderer
    .create(<Controls locked={true} closed={true}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Control button is toggled for unlocked and open', () => {
    const tree = renderer
      .create(<Controls locked={false} closed={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});