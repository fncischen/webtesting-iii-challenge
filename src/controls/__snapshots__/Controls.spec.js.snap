// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Control button is toggled for locked and closed 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={false}
  >
    Unlock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Open Gate
  </button>
</div>
`;

exports[`Controls Control button is toggled for unlocked and open 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={false}
  >
    Close Gate
  </button>
</div>
`;
